files/usr/share/jeos-firstboot/raspberrywifi

```diff
@@ -0,0 +1,138 @@
+        list+=("${line}" '')
+    done < <(ip link set $wlan_device up && iwlist $wlan_device scan|grep ESSID|cut -d':' -f2|cut -d'"' -f2|sort -u)
+    [ -n "${list[*]}" ]
+}
+
+raspberrywifi_wlan_error()
+{
+    d --title $"Error" --msgbox "$1" 0 0
+    dialog --yesno $"Do you want to retry?" 0 0
+}
+
+# This is called by jeos-firstboot for user
+# interaction and access to the global systemd_firstboot_args array
+raspberrywifi_systemd_firstboot()
+{
+    if is_raspberry && ls -d /sys/class/net/*/wireless &>/dev/null; then
+        if dialog --yesno $"Configure wireless network?" 0 0; then
+            config_wireless=true
+        fi
+    fi
+    $config_wireless || return
+
+    while true
+    do
+        if raspberrywifi_get_wlan_devices && d --menu  $"Select Wireless card to configure" 0 0 $dh_menu "${list[@]}"; then
+            wlan_device="$result"
+        else
+            if raspberrywifi_wlan_error $"Error listing wlan devices"; then
+                continue
+            else
+                break
+            fi
+        fi
+
+        if raspberrywifi_get_wlan_networks && d --menu  $"Select Wireless network to connect" 0 0 $dh_menu "${list[@]}"; then
+            wlan_network="$result"
+        else
+            if wlan_error $"Error listing wireless networks"; then
+                continue
+            else
+                break
+            fi
+        fi
+
+        if d --menu $"Select authentication mode" 0 0 $dh_menu $"WPA-PSK" '' $"WPA-EAP" '' $"Open" ''; then
+            wlan_auth_mode="$result"
+        fi
+        wlan_auth_mode_conf=
+        if [ "$wlan_auth_mode" = "WPA-EAP" ]; then
+            wlan_auth_mode_conf=eap
+            if d --inputbox $"Username" 0 0; then
+                wlan_username="$result"
+            fi
+        fi
+        if [ "$wlan_auth_mode" = "WPA-PSK" ]; then
+            wlan_auth_mode_conf=psk
+        fi
+        if [ "$wlan_auth_mode" = "Open" ]; then
+            wlan_auth_mode_conf=open
+        fi
+        if [ "$wlan_auth_mode" != "Open" ]; then
+            wlan_password=
+            if d --insecure --passwordbox  $"Password" 0 0; then
+                wlan_password="$result"
+            fi
+        fi
+        break
+    done
+
+    config_file=`mktemp -qt 'firstboot-XXXXXX'`
+
+    cat << EOF > $config_file
+BOOTPROTO='dhcp'
+STARTMODE='auto'
+WIRELESS_AP_SCANMODE='1'
+WIRELESS_AUTH_MODE='$wlan_auth_mode_conf'
+WIRELESS_ESSID='$wlan_network'
+WIRELESS_MODE='Managed'
+EOF
+
+    if [ $wlan_auth_mode = "WPA-PSK" ]; then
+        echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
+    fi
+
+    if [ $wlan_auth_mode = "WPA-EAP" ]; then
+        echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
+        echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
+        echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
+    fi
+
+    run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+}
+
+# This is called by jeos-firstboot for user
```

Comment:
Nope

Reply:
Nope for comment?

Reply:
Yup - the comment for `_post` should be different than the one for `_systemd_firstboot`.

Reply:
Ok now there is very cryptic comment

Reply:
This comment doesn't make much sense. What about

```
# This is called after the configuration steps finished successfully
```